### PR TITLE
CNDIT 1288 : Fix the Patient/Provider Name processing. 

### PR DESCRIPTION
**Pull request by alog-enquizit**
For patient/provider with multiple child records this is the processing logic:

- Go through the name records
- Select the record with the max person_uid
- Select the one with the max seq id

Added unit tests to validate the logic.

### CODE REVIEW

File: data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/transformer/DataPostProcessor.java [hunk 1]
```diff
@@ -10,20 +10,34 @@
...
 @Slf4j
 @NoArgsConstructor
 public class DataPostProcessor {
     UtilHelper utilHelper = UtilHelper.getInstance();
 
     public <T extends PersonExtendedProps> void processPersonName(String name, T pf) {
         if (!ObjectUtils.isEmpty(name)) {
-            Arrays.stream(utilHelper.deserializePayload(name, Name[].class))
+            List<Name> nameList = Arrays.stream(utilHelper.deserializePayload(name, Name[].class))
                     .filter(pName -> !ObjectUtils.isEmpty(pName.getPersonUid()))
```
Comment (upasanapattnaik-eq) [inline data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/transformer/DataPostProcessor.java:29]: Adding in a comment for integration of nm_use_cd in logic.

File: data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/transformer/DataPostProcessor.java [hunk 2]
```diff
@@ -10,20 +10,34 @@
...
-                    .max(Comparator.comparing(Name::getPersonUid))
-                    .map(n -> n.updatePerson(pf));
+                    .collect(groupingBy(Name::getPersonUid, TreeMap::new, toList()))
+                    .lastEntry()
```
Comment (sveselev) [inline data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/transformer/DataPostProcessor.java:31]: 👍👍👍

**Review by sveselev — APPROVED** [review]
lgtm!

File: data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/transformer/DataPostProcessor.java (hunk 1 above)
Comment (alog-enquizit) [reply data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/transformer/DataPostProcessor.java]: Updated the logic for the `nm_use_cd`
It is applicable only for the Patient i.e., Legal and alias.
For provider only Legal names.

**Review by upasanapattnaik-eq — APPROVED** [review]
Looks good! It looks like some of the UIDs in D_Patient are not aligned with this logic and it is likely that they got missed during a classic ETL run. 
Post 2023, the values are aligned with this logic. I validated with a couple of uids after 2023-01-01 ETL run. Thank you for implementing the updates!

**Pull request merged**